src/components/PaymentInfoLine.js

```diff
@@ -0,0 +1,78 @@
+/* global Nimiq */
+/* global AddressInfo */
+/* global Timer */
+/* global TemplateTags */
+/* global NumberFormatting */
+
+class PaymentInfoLine { // eslint-disable-line no-unused-vars
+    /**
+     * @param {object}           paymentInfo
+     * @param {string}           paymentInfo.recipient
+     * @param {string|undefined} [paymentInfo.label]
+     * @param {URL|undefined}    [paymentInfo.imageUrl]
+     * @param {number}           paymentInfo.lunaAmount
+     * @param {number|undefined} [paymentInfo.fiatAmount]
+     * @param {string|undefined} [paymentInfo.fiatCurrency]
+     * @param {number|undefined} [paymentInfo.time]
+     * @param {number|undefined} [paymentInfo.expires]
+     * @param {HTMLElement}      [$el]
+     */
+    constructor(paymentInfo, $el) {
+        this.$el = PaymentInfoLine._createElement($el);
+
+        /** @type HTMLElement */
+        const $nimAmount = (this.$el.querySelector('.nim-amount'));
+        const nimAmount = NumberFormatting.formatNumber(Nimiq.Policy.lunasToCoins(paymentInfo.lunaAmount));
+        $nimAmount.textContent = `${nimAmount} NIM`;
+
+        /** @type HTMLElement */
+        const $fiatAmount = (this.$el.querySelector('.fiat-amount'));
+        if (paymentInfo.fiatAmount !== undefined && paymentInfo.fiatCurrency !== undefined) {
```

Comment:
Can `fiatAmount` also be `null` and would that be handled differently? Otherwise I think `fiatAmount` can not be `0` (should be validated in the RequestParser), thus we can use the simple truthy check here instead:

```suggestion
        if (paymentInfo.fiatAmount && paymentInfo.fiatCurrency) {
```

Reply:
`fiatAmount` can be `number|undefined`. The check for `undefined` was just to make this explicit.
But I can change it.